Implementar sugestões do pylint para o arquivo Graph.py

(venv) ➜  search git:(main) ✗ pylint Graph.py 
************* Module aigyminsper.search.Graph
Graph.py:28:78: C0303: Trailing whitespace (trailing-whitespace)
Graph.py:31:0: C0303: Trailing whitespace (trailing-whitespace)
Graph.py:43:0: C0303: Trailing whitespace (trailing-whitespace)
Graph.py:44:0: C0303: Trailing whitespace (trailing-whitespace)
Graph.py:65:0: C0303: Trailing whitespace (trailing-whitespace)
Graph.py:72:0: C0303: Trailing whitespace (trailing-whitespace)
Graph.py:79:0: C0303: Trailing whitespace (trailing-whitespace)
Graph.py:98:0: C0304: Final newline missing (missing-final-newline)
Graph.py:1:0: C0114: Missing module docstring (missing-module-docstring)
Graph.py:1:0: C0103: Module name "Graph" doesn't conform to snake_case naming style (invalid-name)
Graph.py:16:11: C0121: Comparison 'self.father_node == None' should be 'self.father_node is None' (singleton-comparison)
Graph.py:27:8: R1705: Unnecessary "else" after "return", remove the "else" and de-indent the code inside it (no-else-return)
Graph.py:27:11: C0121: Comparison 'self.father_node != None' should be 'self.father_node is not None' (singleton-comparison)
Graph.py:64:8: W0107: Unnecessary pass statement (unnecessary-pass)
Graph.py:71:8: W0107: Unnecessary pass statement (unnecessary-pass)
Graph.py:78:8: W0107: Unnecessary pass statement (unnecessary-pass)
Graph.py:85:8: W0107: Unnecessary pass statement (unnecessary-pass)
Graph.py:91:19: E1101: Instance of 'State' has no 'operator' member (no-member)
Graph.py:98:8: W0107: Unnecessary pass statement (unnecessary-pass)

-----------------------------------
Your code has been rated at 2.58/10

Now, the status is: 

pylint Graph.py                                 
************* Module aigyminsper.search.Graph
Graph.py:1:0: C0103: Module name "Graph" doesn't conform to snake_case naming style (invalid-name)
Graph.py:88:19: E1101: Instance of 'State' has no 'operator' member (no-member)

------------------------------------------------------------------
Your code has been rated at 7.69/10 (previous run: 7.69/10, +0.00)

--- a/aigyminsper/search/Graph.py
+++ b/aigyminsper/search/Graph.py
@@ -57,6 +57,9 @@ class State(ABC):
     - print: returns a string with the operator that generated the current state
     """
 
+    def __init__(self, operator=None):
+        self.operator = operator
+
     @abstractmethod
     def successors(self):
         """

--- a/aigyminsper/search/SearchAlgorithms.py
+++ b/aigyminsper/search/SearchAlgorithms.py
@@ -1,5 +1,5 @@
 from collections import deque
-from aigyminsper.search.Graph import Node
+from aigyminsper.search.graph import Node
 
 def sortFunction(val):
     """

--- a/aigyminsper/search/__init__.py
+++ b/aigyminsper/search/__init__.py
@@ -13,7 +13,7 @@ Exposes the main classes of the search module:
 - SubidaMontanhaEstocastico: the class that implements the Stochastic Hill Climbing search algorithm.
 """
 
-from .Graph import State
+from .graph import State
 from .SearchAlgorithms import (
     SearchAlgorithm,
     BuscaLargura,

--- a/docs/src/Puzzle8.py
+++ b/docs/src/Puzzle8.py
@@ -1,6 +1,6 @@
 from unittest import result
 from aicode.search.SearchAlgorithms import AEstrela
-from aicode.search.Graph import State
+from aicode.search.graph import State
 import math
 
 class Puzzle8(State):

--- a/docs/src/U2.py
+++ b/docs/src/U2.py
@@ -1,5 +1,5 @@
 from aicode.search.SearchAlgorithms import BuscaCustoUniforme
-from aicode.search.Graph import State
+from aicode.search.graph import State
 
 class U2(State):
 

--- a/docs/src/VacuumWorldGeneric.py
+++ b/docs/src/VacuumWorldGeneric.py
@@ -1,5 +1,5 @@
 from aigyminsper.search.SearchAlgorithms import AEstrela, BuscaGananciosa, BuscaLargura, BuscaProfundidade, BuscaProfundidadeIterativa
-from aigyminsper.search.Graph import State
+from aigyminsper.search.graph import State
 import numpy
 import sys
 

--- a/docs/user_guide.md
+++ b/docs/user_guide.md
@@ -3,7 +3,7 @@
 In order to create an agent you must implement the `State` interface as shown below:
 
 ```python
-from aigyminsper.search.Graph import State
+from aigyminsper.search.graph import State
 
 class MyAgent(State):
 
@@ -12,24 +12,24 @@ class MyAgent(State):
         self.operator = op
         # You must define how to represent the state
         #TODO
-    
+
     def successors(self):
         successors = []
         #TODO
         # you must define how to generate the successors for each operator (action)
         return successors
-    
+
     def is_goal(self):
         # You must define the goal state
         pass
-    
+
     def description(self):
         return "Problem description"
-    
+
     def cost(self):
         # Return the cost of each operator (action)
         return 1
-    
+
     def env(self):
         #
         # This methos is used to return a description of the state (environment).
@@ -38,9 +38,9 @@ class MyAgent(State):
         None
 ```
 
-You, as a developer, must implement the methods `successors`, `is_goal`, `description`, `cost`, and `env`, and describe how the world must be represented. 
+You, as a developer, must implement the methods `successors`, `is_goal`, `description`, `cost`, and `env`, and describe how the world must be represented.
 
-The next step is define the best algorithm to solve the problem. 
+The next step is define the best algorithm to solve the problem.
 
 ```python
 from aigyminsper.search.SearchAlgorithms import BuscaLargura

--- a/tests/AspiradorPo.py
+++ b/tests/AspiradorPo.py
@@ -1,5 +1,5 @@
 from aigyminsper.search.SearchAlgorithms import BuscaProfundidadeIterativa
-from aigyminsper.search.Graph import State
+from aigyminsper.search.graph import State
 
 class AspiradorPo(State):
 

--- a/tests/Equacao.py
+++ b/tests/Equacao.py
@@ -1,5 +1,5 @@
 from aigyminsper.search.CSPAlgorithms import SubidaMontanha, SubidaMontanhaEstocastico
-from aigyminsper.search.Graph import State
+from aigyminsper.search.graph import State
 import numpy as np
 
 class Equacao(State):

--- a/tests/ProblemSpecificationExample.py
+++ b/tests/ProblemSpecificationExample.py
@@ -1,5 +1,5 @@
 from aigyminsper.search.SearchAlgorithms import BuscaProfundidadeIterativa
-from aigyminsper.search.Graph import State
+from aigyminsper.search.graph import State
 
 class ProblemSpecification(State):
 

--- a/tests/SumOne.py
+++ b/tests/SumOne.py
@@ -1,5 +1,5 @@
 from aigyminsper.search.SearchAlgorithms import BuscaLargura
-from aigyminsper.search.Graph import State
+from aigyminsper.search.graph import State
 
 class SumOne(State):
 

--- a/tests/poi.py
+++ b/tests/poi.py
@@ -1,4 +1,4 @@
-from aigyminsper.search.Graph import State
+from aigyminsper.search.graph import State
 from aigyminsper.search.SearchAlgorithms import BuscaLargura, BuscaProfundidadeIterativa
 from aigyminsper.search.SearchAlgorithms import BuscaProfundidade,BuscaCustoUniforme
 

